app/controllers/budgets_controller.rb

```diff
@@ -1,6 +1,7 @@
 class BudgetsController < ApplicationController
+  before_action :set_grant
```

Comment:
@joedean remember that this is the budget controller not the grant controller.  Every action for the budget needs the grant.